modules/detectron/CMakeLists.txt

```diff
@@ -33,7 +33,7 @@ if(BUILD_CAFFE2_OPS)
     target_link_libraries(caffe2_detectron_ops_hip torch)
     install(TARGETS caffe2_detectron_ops_hip DESTINATION lib)
   elseif(NOT IOS_PLATFORM)
-    add_library(caffe2_detectron_ops SHARED ${Detectron_CPU_SRCS})
+    add_library(caffe2_detectron_ops ${Detectron_CPU_SRCS})
```

Comment:
Can you make this change conditional on `BUILD_SHARED_LIBS` environment variable?

Reply:
Yes, I can do that, but there is no need to add a conditional branch.

`BUILD_SHARED_LIBS` is a global flag in CMake. According to CMake documentation, if no `STATIC` or `SHARED` is given to `add_library`, the library type will be based on whether the value of `BUILD_SHARED_LIBS` is `ON`.

- https://cmake.org/cmake/help/latest/command/add_library.html#normal-libraries
- https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Reply:
Hi @malfet , not sure if my previous comment made sense?